util/rom-ext-manifest-generator.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import copy
+from pathlib import Path
+
+import hjson
+from mako.template import Template
+
+from topgen.c import MemoryRegion, Name
+
+DESC = """TODO rom-ext-manifest-gen"""
+
+USAGE = '''
+  rom-ext-manifest-generator [options]
+  rom-ext-manifest-generator [options] <input>
+  rom-ext-manifest-generator (-h | --help)
+'''
+
+# This file is $REPO_TOP/util/rom-ext-manifest-generator.py, so it takes two
+# parent() calls to get back to the top.
+REPO_TOP = Path(__file__).resolve().parent.parent
+
+
+# Adds `size_in_bytes` entry to every parsed ROM_EXT manifest field.
+# fields[N]    =>    fields[N]
+#   - ...              - ...
+#   - size             - size
+#                      - size_in_bytes
+def add_size_in_bytes(fields):
+    for field in fields:
+        field['size_in_bytes'] = str(int(field['size'], 0) // 8)
+
+
+# Adds `offset_in_bytes` entry to every parsed ROM_EXT manifest field.
+# fields[N]    =>    fields[N]
+#   - ...              - ...
+#   - size_in_bytes    - size_in_bytes
+#                      - offset_in_bytes
+def add_offsets_in_bytes(fields):
+    accumulated_offset = 0x0
+
+    for field in fields:
+        field['offset'] = str(accumulated_offset)
+        accumulated_offset += int(field['size_in_bytes'], 0)
+
+
+# Removes `reserved` entries to from parsed ROM_EXT manifest fields.
+# field0, reserved, field1, ... => field0, field1, ...
+# This is simple way to prevent `generate_cheader` from generating
+# defines for metadata.
+def strip_reserved(fields):
+    for field in fields:
+        if field['name'] == "reserved":
+            fields.remove(field)
+
+
+# Generates C header file from the `template_file`. It produces a dictionary
+# where `fields`, where a field name is a key, and the `MemoryRegion` object
+# is a value. This dictionary is used in the `template_file`, and the resulting
+# header file is placed in the `output_dir`.
+def generate_cheader(fields, template_file, output_dir):
+    base_name = Name.from_snake_case("ROM_EXT")
+
+    items = [(field['name'],
+              MemoryRegion(base_name + Name.from_snake_case(field['name']),
+                           field['offset'], field['size_in_bytes']))
+             for field in fields]
```

Comment:
Nit: this is slightly hard to read, so maybe it aught to be a for loop instead?

Reply:
Done.